crates/bevy_pbr/src/deferred/pbr_deferred_functions.wgsl

```diff
@@ -0,0 +1,146 @@
+#define_import_path bevy_pbr::pbr_deferred_functions
+#import bevy_pbr::pbr_types PbrInput, standard_material_new, STANDARD_MATERIAL_FLAGS_FOG_ENABLED_BIT, STANDARD_MATERIAL_FLAGS_UNLIT_BIT
+#import bevy_pbr::pbr_deferred_types as deft
+#import bevy_pbr::pbr_functions as pbr_functions
+#import bevy_pbr::rgb9e5 as rgb9e5
+#import bevy_pbr::mesh_view_bindings as view_bindings
+
+//TODO Griffin This doesn't seem to work
+//#ifdef NORMAL_PREPASS_OR_DEFERRED_PREPASS 
+//#import bevy_pbr::prepass_bindings view
+//#else
+//#import bevy_pbr::mesh_view_bindings view
+//#endif // NORMAL_PREPASS_OR_DEFERRED_PREPASS
+
+// TODO Griffin using mesh_view_bindings here because of:
+//  ┌─ bevy_pbr\src\prepass\prepass_bindings.wgsl:7:1
+//  │
+//7 │ var<uniform> view: bevy_render::view::View;
+//  │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^ naga::GlobalVariable [36]
+//  │
+//  = Bindings for [36] conflict with other resource
+#import bevy_pbr::mesh_view_bindings view
+
+// ---------------------------
+// from https://github.com/DGriffin91/bevy_coordinate_systems/blob/main/src/transformations.wgsl
+// ---------------------------
+
+/// Convert a ndc space position to world space
+fn position_ndc_to_world(ndc_pos: vec3<f32>) -> vec3<f32> {
+    let world_pos = view.inverse_view_proj * vec4(ndc_pos, 1.0);
+    return world_pos.xyz / world_pos.w;
+}
+
+/// Convert ndc space xy coordinate [-1.0 .. 1.0] to uv [0.0 .. 1.0]
+fn ndc_to_uv(ndc: vec2<f32>) -> vec2<f32> {
+    return ndc * vec2(0.5, -0.5) + vec2(0.5);
+}
+
+/// Convert uv [0.0 .. 1.0] coordinate to ndc space xy [-1.0 .. 1.0]
+fn uv_to_ndc(uv: vec2<f32>) -> vec2<f32> {
+    return (uv - vec2(0.5)) * vec2(2.0, -2.0);
+}
+
+/// Returns the (0.0, 0.0) .. (1.0, 1.0) position within the viewport for the current render target.
+/// [0 .. render target viewport size] eg. [(0.0, 0.0) .. (1280.0, 720.0)] to [(0.0, 0.0) .. (1.0, 1.0)]
+fn frag_coord_to_uv(frag_coord: vec2<f32>) -> vec2<f32> {
+    return (frag_coord - view.viewport.xy) / view.viewport.zw;
+}
+
+/// Convert frag coord to ndc.
+fn frag_coord_to_ndc(frag_coord: vec4<f32>) -> vec3<f32> {
+    return vec3(uv_to_ndc(frag_coord_to_uv(frag_coord.xy)), frag_coord.z);
+}
```

Comment:
Agreed. As far as I'm concerned we can add everything from https://github.com/DGriffin91/bevy_coordinate_systems/blob/main/src/transformations.wgsl I've been using this regularly for quite a while now and love it. 

Shouldn't be a part of this PR though.

Reply:
@superdump https://github.com/bevyengine/bevy/pull/9726

Reply:
I've re-added that PR to the 0.12 milestone as a dependency, but I won't block merging this PR based on the other one being merged as it will be quick to refactor.